drivers/storage/portworx/util/util.go

```diff
@@ -1528,10 +1530,12 @@ func ShouldUseClusterDomain(node *api.StorageNode) (bool, error) {
 // Get list of storagenodes that are a part of the current cluster that need a node PDB
 func NodesNeedingPDB(k8sClient client.Client, nodeEnumerateResponse *api.SdkNodeEnumerateWithFiltersResponse, k8sNodeList *v1.NodeList) ([]string, error) {
 
-	// Get list of kubernetes nodes that are a part of the current cluster
+	// Get list of kubernetes nodes that are a part of the current cluster and are not cordoned
 	k8sNodesStoragePodCouldRun := make(map[string]bool)
 	for _, node := range k8sNodeList.Items {
-		k8sNodesStoragePodCouldRun[node.Name] = true
+		if !node.Spec.Unschedulable {
```

Comment:
The actual reason I didnt create PDB for cordoned nodes is because it creates it by default with minAvailable 1. And if a node is cordoned then it should have had a PDB of 1 initially so there is no need to update that value again. If its not selected for upgrade then its PDB remains, if it is selected then changes to 1. I felt there was no need to update the PDB for cordoned nodes as they will anyways have one

Reply:
I understood your original reason, but I was thinking of the case where for some reason the PDB got deleted.

As I explained above, this is a good backdoor to forcefully let k8s upgrade on a particular node if user understands the risk or if there is a bug in operator that is blocking the upgrade and we need a temporary workaround to unblock the upgrade.